compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -1683,7 +1683,7 @@ fn assemble_candidates_from_impls<'cx, 'tcx>(
                                 &obligation.with(
                                     selcx.tcx(),
                                     ty::Binder::dummy(
-                                        selcx.tcx().at(obligation.cause.span()).mk_trait_ref(LangItem::Sized, [self_ty]),
+                                        ty::TraitRef::from_lang_item(selcx.tcx().at(obligation.cause.span()), LangItem::Sized, [self_ty]),
                                     )
                                     .without_const(),
```

Comment:
annoying that we have to keep the binder for `without_const`. Could imagine adding the same method to `TraitRef` as well.

Reply:
I'm not sure about the naming, but here we go: 1b8c778